alacritty/src/migrate/mod.rs

```diff
@@ -123,9 +123,58 @@ fn migrate_toml(toml: String) -> Result<DocumentMut, String> {
     move_value(&mut document, &["import"], &["general", "import"])?;
     move_value(&mut document, &["shell"], &["terminal", "shell"])?;
 
+    migrate_mouse_hints(&mut document)?;
+
     Ok(document)
 }
 
+/// replaces hints.enabled.mouse.enabled by hints.enabled.mouse.button, changing the
+/// content to match
+fn migrate_mouse_hints(document: &mut DocumentMut) -> Result<(), String> {
+    fn map_hints_mouse(hints_mouse_enabled: Item) -> Item {
+        match hints_mouse_enabled {
+            // = true
+            Item::Value(v) if v.is_bool() && v.as_bool().unwrap() => {
+                Item::Value(toml_edit::Value::from("Left"))
+            },
+            // = false
+            Item::Value(v) if v.is_bool() && !v.as_bool().unwrap() => {
+                Item::Value(toml_edit::Value::from("None"))
+            },
+            other => other,
+        }
+    }
+
+    let table =
+        document.as_item_mut().as_table_like_mut().expect("Moving from unsupported TOML structure");
+    let hints_table = match table.get_key_value_mut("hints") {
+        Some((_, item)) => item.as_table_like_mut().expect("'hints' is expected to be a table"),
+        None => return Ok(()),
+    };
+    let enabled_hints = match hints_table.get_key_value_mut("enabled") {
+        Some((_, item)) => item
+            .as_array_of_tables_mut()
+            .expect("'hints.enabled' is expected to be an array of tables"),
+        None => return Ok(()),
+    };
+
+    for hint in enabled_hints.iter_mut() {
+        let hint_mouse_table = match hint.get_key_value_mut("mouse") {
+            Some((_, mouse)) => {
+                mouse.as_table_like_mut().expect("'mouse' is expected to be a table")
+            },
+            None => continue,
+        };
+        let value_enabled = match hint_mouse_table.remove("enabled") {
+            Some(value) => value,
+            None => continue,
+        };
+        hint_mouse_table.insert("button", map_hints_mouse(value_enabled));
+    }
```

Comment:
All of this essentially just seems to duplicate the purpose of `move_value`. So this really should be implemented in the proper function so it can be reused in the future.